x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/inference/RestStartTrainedModelDeploymentAction.java

```diff
@@ -79,49 +80,47 @@ protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient
             );
         } else {
             request = new StartTrainedModelDeploymentAction.Request(modelId, deploymentId);
-            if (restRequest.hasParam(TIMEOUT.getPreferredName())) {
-                TimeValue openTimeout = restRequest.paramAsTime(
-                    TIMEOUT.getPreferredName(),
-                    StartTrainedModelDeploymentAction.DEFAULT_TIMEOUT
-                );
-                request.setTimeout(openTimeout);
-            }
-            request.setWaitForState(
-                AllocationStatus.State.fromString(restRequest.param(WAIT_FOR.getPreferredName(), AllocationStatus.State.STARTED.toString()))
-            );
-            RestCompatibilityChecker.checkAndSetDeprecatedParam(
-                NUMBER_OF_ALLOCATIONS.getDeprecatedNames()[0],
-                NUMBER_OF_ALLOCATIONS.getPreferredName(),
-                RestApiVersion.V_8,
-                restRequest,
-                (r, s) -> r.paramAsInt(s, request.getNumberOfAllocations()),
-                request::setNumberOfAllocations
-            );
-            RestCompatibilityChecker.checkAndSetDeprecatedParam(
-                THREADS_PER_ALLOCATION.getDeprecatedNames()[0],
-                THREADS_PER_ALLOCATION.getPreferredName(),
-                RestApiVersion.V_8,
-                restRequest,
-                (r, s) -> r.paramAsInt(s, request.getThreadsPerAllocation()),
-                request::setThreadsPerAllocation
-            );
-            request.setQueueCapacity(restRequest.paramAsInt(QUEUE_CAPACITY.getPreferredName(), request.getQueueCapacity()));
-            if (restRequest.hasParam(CACHE_SIZE.getPreferredName())) {
-                request.setCacheSize(
-                    ByteSizeValue.parseBytesSizeValue(restRequest.param(CACHE_SIZE.getPreferredName()), CACHE_SIZE.getPreferredName())
-                );
-            } else if (defaultCacheSize != null) {
-                request.setCacheSize(defaultCacheSize);
-            }
-            request.setQueueCapacity(restRequest.paramAsInt(QUEUE_CAPACITY.getPreferredName(), request.getQueueCapacity()));
-            request.setPriority(
-                restRequest.param(
-                    StartTrainedModelDeploymentAction.TaskParams.PRIORITY.getPreferredName(),
-                    request.getPriority().toString()
-                )
+        }
```

Comment:
The only change here is that I moved this closing brace bracket to here so that all of the below code runs regardless of whether or not there is a body in the request